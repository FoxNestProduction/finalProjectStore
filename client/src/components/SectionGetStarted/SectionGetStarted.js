import React, { memo } from 'react';
import { Typography, Container, Box, Rating, Button } from '@mui/material';
import { Link as RouterLink } from 'react-router-dom';
import useBreakpoint from '../../customHooks/useBreakpoint';
import { stylesReviewAmount, stylesStarts, stylesTrustpilot, stylesProBnt, stylesStartedBnt, stylesTextContainer, stylesLabel, stylesTitle, stylesSection, stylesDescription, stylesActions, stylesRating, stylesImage } from './styles';

const SectionGetStarted = () => {
  const starsValue = 4;
  const breakPoint = useBreakpoint();
  const title = {
    short:
  <>
    Enjoy Foods
    <br />
    Over World
  </>,
    long:
  <>
    Enjoy Foods All
    <br />
    Over The World
  </>,
  };

  return (
    <Container
      component="section"
      sx={stylesSection}
    >
      <Box sx={stylesTextContainer}>
        <Typography
          component="span"
          sx={stylesLabel}
        >
          OVER 1000 USERS
        </Typography>
        <Typography
          variant="h1"
          component="h1"
          sx={stylesTitle}
        >
          {breakPoint === 'mobile' ? title.short : title.long}
        </Typography>
        <Typography component="p" sx={stylesDescription}>
          {/* eslint-disable-next-line max-len */}
          Eatly helps you set saving goals, earn cash back offers, and get paychecks up to two days early. Get a $20 bonus.
        </Typography>
        <Box sx={stylesActions}>
          <Button sx={stylesStartedBnt} component={RouterLink} to="/menu" variant="contained">
            Get Started
          </Button>
          <Button sx={stylesProBnt} component={RouterLink} to="/pricing" variant="outlined" disabled>
            Go Pro
          </Button>
        </Box>
        <Box sx={stylesRating}>
          <img
            style={{
              position: 'relative',
              top: '-4px',
              left: '-2px',
            }}
            src="/img/layout/trustpilotIcon.png"
            alt=""
          />
          <Typography sx={stylesTrustpilot} variant="h6" component="span">Trustpilot</Typography>
          <Rating sx={stylesStarts} name="read-only" value={starsValue} readOnly />
          <Typography sx={stylesReviewAmount} component="span">4900+</Typography>
        </Box>
      </Box>
      <Box sx={stylesImage}>
        <img
          src="/img/layout/getStartedSectionImage.png"
          alt=""
        />
      </Box>
    </Container>
  );
};

export default memo(SectionGetStarted);
