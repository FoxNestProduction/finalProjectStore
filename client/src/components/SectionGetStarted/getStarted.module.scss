@import "../../styles/mixins.scss";
@import "../../styles/variables.scss";

.getStarted {
  min-width: 320px;


  @media (min-width: $bp-tablet) {


    font-size: 12px;
  }

  @media (min-width: $bp-desktop) {

  }
  &__title {
    @media (min-width: $bp-tablet) {


      font-size: 12px;
    }

    @media (min-width: $bp-desktop) {

    }
  }
}

