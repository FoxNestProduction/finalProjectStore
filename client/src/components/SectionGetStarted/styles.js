const buttonSize = {
  height: {
    mobile: '41px',
    tablet: '47px',
    desktop: '60px',
  },
  width: {
    mobile: '110px',
    tablet: '110px',
    desktop: '188px',
  },
  fontSize: {
    mobile: '14px',
    tablet: '14px',
    desktop: '24px',
  },
};

export const stylesSection = {
  display: 'flex',
  flexDirection: { mobile: 'column', tablet: 'row' },
  gap: 1,
  textAlign: {
    mobile: 'center',
    tablet: 'left',
  },
  marginBottom: {
    mobile: '75px',
    tablet: '60px',
    desktop: '66px',
  },
  pt: '40px',
  transition: '.5s',
};

export const stylesTextContainer = {
  display: 'flex',
  flexDirection: 'column',
  width: {
    mobile: '100%',
    tablet: '45%',
    desktop: '52%',
  },
};

export const stylesLabel = {
  color: 'text.secondary',
  fontWeight: 600,
  fontFamily: 'Poppins',
  fontSize: {
    mobile: '9px',
    tablet: '12px',
    desktop: '14px',
  },
  marginBottom: {
    mobile: '5px',
    tablet: '12px',
    desktop: '27px',
  },
  opacity: '.7',
};

export const stylesTitle = {
  minWidth: {
    mobile: '100%',
    tablet: '350px',
  },
  color: 'text.primary',
  fontWeight: 500,
  fontFamily: 'Poppins',
  marginBottom: {
    mobile: '16px',
    tablet: '11px',
    desktop: '32px',
  },
};

export const stylesDescription = {
  color: 'text.primary',
  fontWeight: 400,
  lineHeight: {
    mobile: '17px',
    tablet: '19px',
    desktop: '36px',
  },
  marginBottom: {
    mobile: '41px',
    tablet: '37px',
    desktop: '32px',
  },
  marginLeft: {
    mobile: '10px',
    tablet: '0',
  },
  marginRight: {
    mobile: '10px',
    tablet: '0',
  },
  fontSize: {
    mobile: '12px',
    tablet: '13px',
    desktop: '24px',
  },
};

export const stylesActions = {
  minWidth: '240px',
  marginBottom: {
    mobile: '26px',
    tablet: '23px',
    desktop: '32px',
  },
  display: 'flex',
  justifyContent: {
    mobile: 'center',
    tablet: 'left',
  },
  columnGap: {
    mobile: '14px',
    tablet: '12px',
    desktop: '24px',
  },
};

export const stylesStartedBnt = {
  borderRadius: 'shape.borderRadius',
  backgroundColor: 'secondary.main',
  fontWeight: 500,
  ...buttonSize,
  '&:hover': {
    backgroundColor: 'secondary.hover',
  },
};

export const stylesProBnt = {
  color: 'text.primary',
  borderRadius: 'shape.borderRadius',
  backgroundColor: 'background.default',
  border: '2px solid #6C5FBC',
  fontWeight: 500,
  ...buttonSize,
  '&:hover': {
    color: 'text.primaryLight',
    backgroundColor: 'primary.hover',
  },
};

export const stylesRating = {
  minWidth: '240px',
  marginBottom: {
    mobile: '47px',
  },
  fontFamily: 'Poppins',
  display: 'flex',
  columnGap: '2px',
  justifyContent: {
    mobile: 'center',
    tablet: 'left',
  },
  alignItems: 'center',
};

export const stylesTrustpilot = {
  color: 'text.primary',
  fontWeight: 600,
};

export const stylesStarts = {
  minWeight: '90px',
  fontSize: '20px',
};

export const stylesReviewAmount = {
  color: 'text.secondary',
  fontSize: '10px',
  position: 'relative',
  top: '1px',
  opacity: '.7',

};

export const stylesImage = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',

};
