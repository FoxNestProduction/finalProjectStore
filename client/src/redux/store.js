import { configureStore, getDefaultMiddleware, combineReducers } from '@reduxjs/toolkit';
// todo: правила для redux-persist
// eslint-disable-next-line import/no-extraneous-dependencies
import storage from 'redux-persist/lib/storage';
// eslint-disable-next-line import/no-extraneous-dependencies
import { persistReducer } from 'redux-persist';
import modalSlice from './slices/modalSlice';
import productsSlice from './slices/productsSlice';
import authorizationSlice from './slices/authorizationSlice';
import userSlice from './slices/userSlice';
import partnersSlice from './slices/partnersSlice';
import searchSlice from './slices/searchSlice';
import errorSlice from './slices/errorSlice';
import reviewsSlice from './slices/reviewsSlice';
import favouriteSlice from './slices/favouriteSlice';

const authPersistConfig = {
  key: 'authorization',
  version: 1,
  storage,
};

const userPersistConfig = {
  key: 'user',
  version: 1,
  storage,
};

const searchPersistConfig = {
  key: 'search',
  version: 1,
  storage,
};
const favouritePersistConfig = {
  key: 'favorites',
  version: 1,
  storage,
};

/*
    todo: - ЗВЕРНІТЬ УВАГУ!
      якщо вам потрібно щоб ваш стейт не зникав після перезавантаження
      то юзайте підхід як authorization і user,
      якщо ж ні то варіант написання як для modal всередині reducer
      Якщо що пишіть @Ihor_Kacher
*/
const reducer = combineReducers({
  error: errorSlice,
  modal: modalSlice,
  partners: partnersSlice,
  authorization: persistReducer(authPersistConfig, authorizationSlice),
  user: persistReducer(userPersistConfig, userSlice),
  products: productsSlice,
  search: persistReducer(searchPersistConfig, searchSlice),
  reviews: reviewsSlice,
  favourites: persistReducer(favouritePersistConfig, favouriteSlice),
});

const middleware = getDefaultMiddleware({
  serializableCheck: false,
});

const store = configureStore({
  reducer,
  middleware,
});

export default store;
