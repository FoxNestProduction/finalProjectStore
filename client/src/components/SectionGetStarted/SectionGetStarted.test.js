import React from 'react';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../../redux/store';
import SectionGetStarted from './SectionGetStarted';

describe('SectionGetStarted component', () => {
  test('should render SectionGetStarted component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <SectionGetStarted />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
