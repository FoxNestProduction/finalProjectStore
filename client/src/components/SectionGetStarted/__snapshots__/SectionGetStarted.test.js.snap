// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionGetStarted component should render SectionGetStarted component 1`] = `
<DocumentFragment>
  <section
    class="MuiContainer-root MuiContainer-maxWidthLg css-vct4wb-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-fhi1jl"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 css-1flcdwx-MuiTypography-root"
      >
        OVER 1000 USERS
      </span>
      <h1
        class="MuiTypography-root MuiTypography-h1 css-2s84mo-MuiTypography-root"
      >
        Enjoy Foods All
        <br />
        Over The World
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 css-3by2ru-MuiTypography-root"
      >
        Eatly helps you set saving goals, earn cash back offers, and get paychecks up to two days early. Get a $20 bonus.
      </p>
      <div
        class="MuiBox-root css-1wajcrl"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ibva2t-MuiButtonBase-root-MuiButton-root"
          href="/menu"
          tabindex="0"
        >
          Get Started
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1b1g7hx-MuiButtonBase-root-MuiButton-root"
          href="/pricing"
          tabindex="-1"
        >
          Go Pro
        </a>
      </div>
      <div
        class="MuiBox-root css-q0lyv9"
      >
        <img
          alt=""
          src="/img/layout/trustpilotIcon.png"
          style="position: relative; top: -4px; left: -2px;"
        />
        <span
          class="MuiTypography-root MuiTypography-h6 css-3efye1-MuiTypography-root"
        >
          Trustpilot
        </span>
        <span
          aria-label="4 Stars"
          class="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-x7x6my-MuiRating-root"
          role="img"
        >
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 css-1g16ti7-MuiTypography-root"
        >
          4900+
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-gmuwbf"
    >
      <img
        alt=""
        src="/img/layout/getStartedSectionImage.png"
      />
    </div>
  </section>
</DocumentFragment>
`;
